app/cache/cache-service.js

```diff
@@ -0,0 +1,32 @@
+const NodeCache = require('node-cache');
+const { Logger } = require('@hmcts/nodejs-logging');
+const logger = Logger.getLogger('CacheService');
+
+class Cache {
+
+  constructor(name, ttlSeconds, checkPeriodSeconds) {
+    this.cache = new NodeCache({ stdTTL: ttlSeconds, checkperiod: checkPeriodSeconds });
+    this.name = name;
+  }
+
+  get(key, storeFunction) {
```

Comment:
I think this API is a bit unconventional and makes code a bit more difficult to understand. I would got for a classic api that provides a separate get method and put method. 
Furthermore some clients of this generic cache would only want to do a get. In that case they would be forced to also pass a "do nothing" store function which complicates the clients code.

Reply:
Added the suggested get/set and rename/comment 👍 

With regards to `synchronized`, there isn't a Node default as such. But in practice for this there could still in theory be a race condition caused if two requests with the same jwt were made in quick succession. Worst case would be that this happens and two calls are made to IDAM, and both end up setting the same value in the cache (i.e. second one overrides the first), so it is not a big impact even if it does. It is probably not worth the effort of introducing some kind of logic given how little this should impact the overall performance improvements.